### ISSUE TITLE
Reserve top-level `$schema` property?

### ISSUE THREAD

**Issue by natemoo-re** [design-tokens/community-group#107]
Visual Studio Code [supports](https://code.visualstudio.com/docs/languages/json#_mapping-in-the-json) a top-level `$schema` property which provides their JSON Language Server with automatic completions, hovers, etc. While it is not part of the [JSON Schema specification](https://json-schema.org/draft/2020-12/json-schema-core.html), it is a feature that I see being used with increasing frequency. See [schemastore.org](https://www.schemastore.org/json) for examples of tools which have public schemas.

IMO calling this property out in the specification would be wise, even just as something that tools should accept but ignore. I've definitely encountered tools that have trouble validating or parsing a JSON file when `$schema` is used and it can be a pain.

**Comment by natemoo-re** [comment]
While we're on the topic of JSON Schema, I started working on a [public schema](https://github.com/tokencss/schema) that aligns with much of the current draft. My current version is a little more opinionated in group structure than the spec, but I'm sure the token definitions are still useful.

I'm happy to contribute to an official spec schema if there's interest!

**Comment by TravisSpomer** [comment]
Definitely agree with this. The token format is currently missing some kind of versioning field anyway, and `$schema` could potentially even be used for that purpose too, though I'm not entirely sure if that's wise.

**Comment by c1rrus** [comment]
I agree this could be useful and could double as a versioning field (something we've been meaning to add anyway). We could have the spec version that the file conforms to as part of the URL to the corresponding schema. E.g.:

```jsonc
{
  "$schema": "https://designtokens.org/spec/v1.2.3/schema.json",
  // ^ This file follows version 1.2.3 of the DTCG spec

  // tokens and groups go here as per usual
}
```

I also agree that having a JSON schema for our spec could be useful. It would allow some basic validation of files and can enable some auto-completion features in some IDEs. It's something that's been suggested before and has been discussed at previous editors meetings too. We've just not gotten around to properly exploring the topic yet. When we do, I'm sure @natemoo-re's schema will be a valuable input, so thanks for sharing that!

Also, since we've now [decided to use `$` as our prefix for all format-specific properties](https://github.com/design-tokens/community-group/issues/61#issuecomment-1021459887), this would fit in nicely with that :-)

**Comment by drwpow** [comment]
You’ll never believe it! Behold, thanks to @dgtlntv:

- `https://www.designtokens.org/schemas/2025.10/format.json`: core format (`.tokens.json`)
- `https://www.designtokens.org/schemas/2025.10/resolver.json`: Resolver files (`.resolver.json`)

We’ll be updating examples / socializing it more, but you heard it here first!
